lib/presentation/book/view/home_page_view.dart

```diff
@@ -9,8 +18,138 @@ class HomePageView extends StatefulWidget {
 }
 
 class _HomeBookPageView extends State<HomePageView> {
+  late final _controller;
+  late List<Book> bookList = [];
+
+  @override
+  void initState() {
+    _controller = PageController(viewportFraction: 0.9, initialPage: 0);
+
+    super.initState();
+  }
+
   @override
   Widget build(BuildContext context) {
-    return Scaffold();
+    return BlocConsumer<HomePageBloc, HomePageState>(
+      listener: (context, state) {
+        if (state is LoadingState) {
+          DialogUtils.showLoadingScreen(context);
+        } else if (state is LoadedState) {
+          Navigator.pop(context);
+        }
+      },
+      builder: (BuildContext context, Object? state) {
+        return Scaffold(
+          appBar: AppBar(
+            leading: IconButton(
+              onPressed: () {},
+              icon: Icon(Icons.exit_to_app),
+            ),
+            centerTitle: true,
+            automaticallyImplyLeading: false,
+            title: Text(AppLocalizations.of(context)!.pick_any_book),
+          ),
+          body: SizedBox(
+            height: 500,
+            child: PageView.builder(
+              pageSnapping: true,
+              controller: _controller,
+              itemCount: bookList.length,
+              itemBuilder: (context, index) =>
+                  _buildBookItem(bookList.elementAt(index)),
+            ),
+          ),
+          floatingActionButton: Visibility(
+            visible: true,
+            child: FloatingActionButton(
+              onPressed: () async {
+                final Book? result = await Navigator.pushNamed<dynamic>(
+                    context, kAddNewBookRoute);
+
+                if (result != null) {
+                  bookList.add(result);
```

Comment:
It's not a good place to add to the list. The list should be added after the refresh